From 79bd52aba84539b6dbb5906413e55ef7e27a30cf Mon Sep 17 00:00:00 2001
From: Valentin Raevsky <valentin@compulab.co.il>
Date: Wed, 22 Dec 2021 17:26:09 +0200
Subject: [PATCH 40/40] compulab: imx7: Add peripheral device support on
 sbc-iot

Add peripheral device support on sbc-iot P13.
A USB Type-A cable is required.

Signed-off-by: Valentin Raevsky <valentin@compulab.co.il>
---
 arch/arm/boot/dts/Makefile                      |  1 +
 arch/arm/boot/dts/imx7d-sbc-iot-imx7-device.dts | 15 +++++++++++++++
 2 files changed, 16 insertions(+)
 create mode 100644 arch/arm/boot/dts/imx7d-sbc-iot-imx7-device.dts

diff --git a/arch/arm/boot/dts/Makefile b/arch/arm/boot/dts/Makefile
index efdb721ffd7e..6c09893d2243 100644
--- a/arch/arm/boot/dts/Makefile
+++ b/arch/arm/boot/dts/Makefile
@@ -698,6 +698,7 @@ dtb-$(CONFIG_SOC_IMX7D) += \
 	imx7d-sbc-imx7-m4.dtb \
 	imx7d-sbc-iot-imx7.dtb \
 	imx7d-sbc-iot-imx7-can.dtb \
+	imx7d-sbc-iot-imx7-device.dtb \
 	imx7d-sbc-iot-imx7-rs485-hdx.dtb \
 	imx7d-sdb.dtb \
 	imx7d-sdb-epdc.dtb \
diff --git a/arch/arm/boot/dts/imx7d-sbc-iot-imx7-device.dts b/arch/arm/boot/dts/imx7d-sbc-iot-imx7-device.dts
new file mode 100644
index 000000000000..f7abbe5ffc87
--- /dev/null
+++ b/arch/arm/boot/dts/imx7d-sbc-iot-imx7-device.dts
@@ -0,0 +1,15 @@
+#include "imx7d-sbc-iot-imx7.dts"
+
+/ {
+	model = "CompuLab CL-SOM-iMX7 on SB-IOT with device support on P13";
+	compatible = "compulab,sbc-iot-imx7", "compulab,cl-som-imx7", "fsl,imx7d";
+};
+
+&usbotg1 {
+	dr_mode = "peripheral";
+	/delete-property/ pinctrl-names;
+	/delete-property/ pinctrl-0;
+	/delete-property/ vbus-supply;
+	status = "okay";
+};
+
-- 
2.17.1

